lib/fluent/plugin/out_newrelic.rb

```diff
@@ -137,6 +122,56 @@ def send_payload(payload)
         handle_response(http.request(request))
       end
 
+      private
+
+      def get_compressed_payloads(logs)
+        return [] if logs.length == 0
+
+        payload = create_payload(logs)
+        compressed_payload = compress(payload)
+
+        if compressed_payload.bytesize < MAX_PAYLOAD_SIZE
```

Comment:
OK, this is reaaaaally picky: should we also support payloads being exactly 1MB in size?

Reply:
Done!

Reply:
Thanks!